src/components/select/AsyncSelect.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import {
+  Select as MantineSelect,
+  SelectProps as MantineSelectProps,
+  SelectItem,
+} from '@mantine/core';
+import { forwardRef, useCallback, useRef } from 'react';
+import { PropsWithLabels } from '~types/utils';
+import { Spinner } from '..';
+import { SelectHighlight } from './Select';
+
+interface Props {
+  className?: string;
+  data: ReadonlyArray<SelectItem>;
+  hasError?: boolean;
+  highlight?: SelectHighlight;
+  id?: string;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  isNotClearable?: boolean;
+  isRequired?: boolean;
+  labelError?: MantineSelectProps['error'];
+  labelNotFound?: MantineSelectProps['nothingFound'];
+  optionComponent?: MantineSelectProps['itemComponent'];
+  onChange?: MantineSelectProps['onChange'];
```

Comment:
Should we also make onChange and value mandatory (for normal select too)? since we never really use these components in forms?